frontend/src/metabase/components/SaveQuestionForm/SaveQuestionForm.tsx

```diff
@@ -90,7 +90,17 @@ export const SaveQuestionForm = ({
               collectionIdFieldName="collection_id"
               dashboardIdFieldName="dashboard_id"
               title={t`Where do you want to save this?`}
-              collectionPickerModalProps={{ models }}
+              collectionPickerModalProps={{
+                models,
+                recentFilter: items =>
+                  items.filter(
+                    item =>
+                      !(
+                        item.model === "collection" ||
+                        item.model === "dashboard"
+                      ) || item.can_write,
+                  ),
```

Comment:
nit: Could we increase the readability a little bit? The rationale for the complex boolean is hard for me to guess. You might consider defining `recentFilter` in the body of the function, as well as slowing down the filter callback with commented lines.

Toying around, maybe something like this could work:
````
// (explanation here)
if (item.can_write) { return true; }

 // (explanation here)
if (item.model === "collection") { return false; }
if (item.model === "dashboard") { return false; }
return true;
```

It's a tad confusing why we include in the list of recents anything writable and also anything that's neither a collection nor a dashboard.